trustpoint/devices/models.py

```diff
@@ -86,13 +86,19 @@ def revoke_ldevid(self: Device, revocation_reason) -> bool:
         if not self.ldevid:
             return False
 
-        if self.device_onboarding_status == Device.DeviceOnboardingStatus.ONBOARDED:
-            self.device_onboarding_status = Device.DeviceOnboardingStatus.REVOKED
-
-        self.ldevid.revoke(revocation_reason)
+        revocation_success = self.ldevid.revoke(revocation_reason)
         self.ldevid = None
+        if self.device_onboarding_status == Device.DeviceOnboardingStatus.ONBOARDED:
+            if revocation_success:
```

Comment:
The chances are probably minuscule, but I think there could be a chance that we revoke the certificate but don't save the state of the device as revoked?
Maybe make the method atomic?